polkadot/node/core/pvf/src/testing.rs

```diff
@@ -47,3 +53,47 @@ pub fn validate_candidate(
 
 	Ok(result)
 }
+
+/// Retrieves the worker paths, checks that they exist and does a version check.
+///
+/// NOTE: This should only be called in dev code (tests, benchmarks) as it relies on the relative
+/// paths of the built workers.
+pub fn get_and_check_worker_paths() -> (PathBuf, PathBuf) {
+	// Only needs to be called once for the current process.
+	static WORKER_PATHS: OnceLock<Mutex<(PathBuf, PathBuf)>> = OnceLock::new();
+	let mutex = WORKER_PATHS.get_or_init(|| {
+		let mut workers_path = std::env::current_exe().unwrap();
+		workers_path.pop();
+		workers_path.pop();
+		let mut prepare_worker_path = workers_path.clone();
+		prepare_worker_path.push(PREPARE_BINARY_NAME);
+		let mut execute_worker_path = workers_path.clone();
+		execute_worker_path.push(EXECUTE_BINARY_NAME);
+
+		// Check that the workers are valid.
+		if !prepare_worker_path.is_executable() || !execute_worker_path.is_executable() {
+			panic!("ERROR: Workers do not exist or are not executable. Workers directory: {:?}", workers_path);
+		}
+
+		let node_version = env!("POLKADOT_NODE_VERSION");
+		let worker_version =
+			get_worker_version(&prepare_worker_path).expect("checked for worker existence");
+		if worker_version != node_version {
+			panic!("ERROR: Prepare worker version {worker_version} does not match node version {node_version}; worker path: {prepare_worker_path:?}");
+		}
+		let worker_version =
+			get_worker_version(&execute_worker_path).expect("checked for worker existence");
+		if worker_version != node_version {
+			panic!("ERROR: Execute worker version {worker_version} does not match node version {node_version}; worker path: {execute_worker_path:?}");
+		}
+
+		// We don't want to check against the commit hash because we'd have to always rebuild
+		// the calling crate on every commit.
+		eprintln!("WARNING: Workers match the node version, but may have changed in recent commits. Please rebuild them if anything funny happens. Workers path: {workers_path:?}");
```

Comment:
You already print this here. So, this doesn't really make any sense to compare the version. If you really want to make it more "safe", you can call here `cargo build` to ensure the worker are there.

Reply:
Well, this warning won't show up unless `-- --nocapture` is passed. I left it because it's better than nothing I guess. I think it's still useful to separately check for worker existence and that they're on the same version.

> If you really want to make it more "safe", you can call here cargo build to ensure the worker are there.

You mean `cargo build` in build.rs, every time the version changes?

Reply:
No, I mean this function does this at "runtime". This is only being used for testing stuff. So, it could ensure that the workers are up to date. It would also make testing easier, as people would not be required to first run `cargo build` after finding out that the workers are not there.